runtime/lib/mirrors_impl.dart

```diff
@@ -1665,7 +1667,10 @@ class _Mirrors {
     return classMirror;
   }
 
-  static TypeMirror reflectType(Type key) {
+  static TypeMirror reflectType(Type key, [List<Type> typeArguments]) {
+    if (typeArguments != null) {
+      key = _resolveType(key, typeArguments);
```

Comment:
Maybe just `return _reflectGenericType(key, typeArguments)` and change the name of `_resolveType (and its return value).
The current code should work, though.